### DISCUSSION TITLE
`where` and `include`: filter if last record satisfies condition?

### DISCUSSION THREAD

**Discussion by ricardo-valero** [prisma/prisma#15496] [Q&A] [answered]
Similar question https://github.com/prisma/prisma/discussions/13556

My schema
```prisma
model Archive {
  id         String    @id @default(cuid())
  date       DateTime  @updatedAt
  value      Boolean   @default(true)
  companies  Company[]
}

model Company {
  id       String    @id @default(cuid())
  date     DateTime  @updatedAt
  name     String    @unique
  archives Archive[]
}
```

Having these two tables, how would someone query all the companies where the last archive is false?

<details>
<summary>Sample data</summary>

```json
{
  "data": {
    "companies": [
      {
        "name": "A",
        "archives": []
      },
      {
        "name": "B",
        "archives": [
          {
            "date": "2022-09-23T20:19:19.821Z",
            "value": true
          }
        ]
      },
      {
        "name": "C",
        "archives": [
          {
            "date": "2022-09-23T20:20:08.521Z",
            "value": true
          },
          {
            "date": "2022-09-23T20:20:11.086Z",
            "value": false
          }
        ]
      },
      {
        "name": "D",
        "archives": [
          {
            "date": "2022-09-23T20:19:33.463Z",
            "value": true
          },
          {
            "date": "2022-09-23T20:19:37.405Z",
            "value": false
          },
          {
            "date": "2022-09-23T20:19:40.937Z",
            "value": true
          }
        ]
      }
    ]
  }
}
```
</details>

Filtering with javascript:
```ts
prisma.company.findMany({
  include: { archives: { take: -1 } },
}).then(
  companies => companies.filter(
    company => company.archives.length === 0
      || company.archives.every(archive => archive.value === false)),
),
```

<details>
<summary>Result</summary>

```json
{
  "data": {
    "companies": [
      {
        "name": "A"
      },
      {
        "name": "C"
      }
    ]
  }
}
```
</details>

Using prisma, my intuition was writing something like this but the result is not what I expect
```ts
prisma.company.findMany({
  where: { archives: { every: { value: false } } },
  include: { archives: { take: -1 } },
}),
```

<details>
<summary>Result</summary>

```json
{
  "data": {
    "companies": [
      {
        "name": "A"
      }
    ]
  }
}
```
</details>

**Comment by austincrim** [comment, marked as answer]
Hey @RicardoValero95 👋,

This operation isn't possible without [$queryRaw](https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access#queryraw), as far as I know. You can also continue to use JavaScript to do the filtering.

Please open a [feature request](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) if you have ideas on how we could handle this in Prisma.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
